cmd/oras/root/attach.go

```diff
@@ -164,7 +169,19 @@ func runAttach(ctx context.Context, opts attachOptions) error {
 
 	root, err := pushArtifact(dst, pack, copy)
 	if err != nil {
-		return err
+		if opts.strict {
+			return err
+		}
+		var re *remote.ReferrersError
+		if !errors.As(err, &re) || !re.IsReferrersIndexDelete() {
+			// not referrer index delete error
+			return err
+		}
+		var errResp *errcode.ErrorResponse
+		if !errors.As(re.Err, &errResp) || errResp.StatusCode != http.StatusMethodNotAllowed {
+			return err
+		}
```

Comment:
Based on the log, this code block can be removed.

Reply:
If this code block is removed, then all non-415 errors occurred during deleting the referrers index will be silently ignored if `--strict` is not set

Is this UX what we want @shizhMSFT?